tests/test_resources.py

```diff
@@ -62,108 +62,98 @@ def test_credit_payment(self):
+                    recurly.Subscription(plan_code = 'gold')
+                ],
+                adjustments = [
+                    recurly.Adjustment(unit_amount_in_cents=1000, description='Item 1',
+                                    quantity=1),
+                    recurly.Adjustment(unit_amount_in_cents=2000, description='Item 2',
+                                    quantity=2),
+                ]
+            )
+        
         with self.mock_request('purchase/invoiced.xml'):
-            collection = purchase.invoice()
+            collection = create_purchase().invoice()
             self.assertIsInstance(collection, InvoiceCollection)
             self.assertIsInstance(collection.charge_invoice, Invoice)
             self.assertIsInstance(collection.credit_invoices, list)
             self.assertIsInstance(collection.credit_invoices[0], Invoice)
             self.assertIsInstance(collection.charge_invoice.line_items[0].shipping_address,
                                   ShippingAddress)
         with self.mock_request('purchase/previewed.xml'):
-            collection = purchase.preview()
+            collection = create_purchase().preview()
             self.assertIsInstance(collection, InvoiceCollection)
             self.assertIsInstance(collection.charge_invoice, Invoice)
         with self.mock_request('purchase/authorized.xml'):
+            purchase = create_purchase()
             purchase.account.email = 'benjamin.dumonde@example.com'
             purchase.account.billing_info.external_hpp_type = 'adyen'
             collection = purchase.authorize()
             self.assertIsInstance(collection, InvoiceCollection)
             self.assertIsInstance(collection.charge_invoice, Invoice)
         with self.mock_request('purchase/captured.xml'):
-            captured_collection = purchase.capture('40625fdb0d71f87624a285476ba7d73d')
+            captured_collection = create_purchase().capture('40625fdb0d71f87624a285476ba7d73d')
             self.assertIsInstance(captured_collection, InvoiceCollection)
             self.assertEquals(captured_collection.charge_invoice.state, 'paid')
         with self.mock_request('purchase/cancelled.xml'):
-            cancelled_collection = purchase.cancel('40625fdb0d71f87624a285476ba7d73d')
+            cancelled_collection = create_purchase().cancel('40625fdb0d71f87624a285476ba7d73d')
             self.assertIsInstance(cancelled_collection, InvoiceCollection)
             self.assertEquals(cancelled_collection.charge_invoice.state, 'failed')
         with self.mock_request('purchase/pending.xml'):
+            purchase = create_purchase()
             purchase.account.email = 'benjamin.dumonde@example.com'
             purchase.account.billing_info.external_hpp_type = 'adyen'
             collection = purchase.pending()
             self.assertIsInstance(collection, InvoiceCollection)
             self.assertIsInstance(collection.charge_invoice, Invoice)
-
-    def test_purchase(self):
```

Comment:
this method test was overwriting the previous (because they have the same name), so I combine both.